examples/cfd/04_burgers.ipynb

```diff
@@ -252,9 +261,10 @@
     "op = Operator([update_u, update_v] + bc_u + bc_v)\n",
     "\n",
     "# Execute the operator for a number of timesteps\n",
-    "op(time=nt, dt=dt, a=nu)\n",
-    "\n",
-    "plot_field(u.data[0])"
+    "for batch_no in range(batches+1):\n",
+    "    op(time=batch_size, dt=dt, a=nu)\n",
```

Comment:
I still don't really like this as it will run `t=0,....,batch_size` at each call. SO it works fine because the wavefield is not saved but it would only recompute the same part of the time history over and over with `save=nt` instead of continuing from previous stop.

Reply:
Is there a way to take a `snapshot` at some point of the procedure?
What do you suggest? Should we maybe only show the last batch?

Reply:
@FabioLuporini @mloubout any thoughts about this?

Reply:
wdym by "procedure" ?

Reply:
The operator runtime.
If we execute for all timesteps we are not able to have a snap for every 100 timesteps for example.

Reply:
I guess this is one of the reasons we keep both batches and timesteps? One could lower the timestep, while suitably increasing the number of batches, to get more frequent snapshots?

Reply:
This is true but if we are fine with what mloubout pointed 4 posts before then OK.

Reply:
ah, I see. Perhaps @mloubout is suggesting that we should rather use ``time_m=... , time_M=...`` instead of ``time=batch_size`` ? this way it would work with save=nt too I suppose? do u see what I mean?

Reply:
I tried to understand what to use from here [TimeFunction Devito Documentation](https://www.devitoproject.org/devito/devito.html?highlight=timefunction#devito.base.TimeFunction)

but I don't quite get why to use save='nt' as it doesn't seem to work...

Reply:
So, I took into account Listing 6 of Devito architecture paper in order to save the state of our wave every `batch_size` timesteps and print this state as an image. My problem is that I cannot find proper documentation for that. Can anyone point me to an example that takes advantage of Listing 6? Additionally, will my method do any additional operations compared to what @mloubout proposes?